models/multiphase/d2q9_pseudo_potential/Dynamics.R

```diff
@@ -0,0 +1,49 @@
+
+# Fluid Density Populations
+AddDensity( name="f[0]", dx= 0, dy= 0, group="f")
+AddDensity( name="f[1]", dx= 1, dy= 0, group="f")
+AddDensity( name="f[2]", dx= 0, dy= 1, group="f")
+AddDensity( name="f[3]", dx=-1, dy= 0, group="f")
+AddDensity( name="f[4]", dx= 0, dy=-1, group="f")
+AddDensity( name="f[5]", dx= 1, dy= 1, group="f")
+AddDensity( name="f[6]", dx=-1, dy= 1, group="f")
+AddDensity( name="f[7]", dx=-1, dy=-1, group="f")
+AddDensity( name="f[8]", dx= 1, dy=-1, group="f")
+
+# Pseudopotential field
+AddField("psi", stencil2d=1, group="pp")
+AddField("neighbour_type", stencil2d=1, group="neighbour_type_group")
+
+# Stages and Actions
+
+# Initialization list
+AddStage("BaseInit"     , "Init", save=Fields$group %in% c("f", "neighbour_type_group"), load=DensityAll$group=="f")
+
+# Iteration list
+AddStage("BaseIteration", "Run"     ,  save=Fields$group %in% c("f", "neighbour_type_group") , load=DensityAll$group %in% c("f","neighbour_type_group"))
+AddStage("PsiIteration" , "calcPsi" ,  save=Fields$name=="psi", load=DensityAll$group %in% c("f"))
+
+AddAction("Init"     , c("BaseInit",      "PsiIteration"))
+AddAction("Iteration", c("BaseIteration", "PsiIteration"))
+
+# Output Values
+AddQuantity( name="U",    unit="m/s", vector=TRUE )
+AddQuantity( name="Ueq",  unit="m/s", vector=TRUE )
+AddQuantity( name="Rho",  unit="kg/m3" )
+AddQuantity( name="Psi",  unit="1" )
+AddQuantity( name="Neighbour_type",  unit="1" )
+AddQuantity( name="F_ff", unit="N", vector=TRUE)
+AddQuantity( name="F_sf", unit="N", vector=TRUE)
+
```

Comment:
Aren't those values debug-only?

Reply:
rather debug, but I think that
```
AddQuantity( name="Rho",  unit="kg/m3" )
AddQuantity( name="Psi",  unit="1" )
AddQuantity( name="F_ff", unit="N", vector=TRUE)
AddQuantity( name="F_sf", unit="N", vector=TRUE)
```
can be keep for tutorial